lib/pure/random.nim

```diff
@@ -162,9 +165,13 @@ proc rand*[T](r: var Rand; a: openArray[T]): T {.deprecated.} =
   ## **Deprecated since v0.20.0:** use ``sample`` instead.
   result = a[rand(r, a.low..a.high)]
 
-proc rand*[T: SomeInteger](t: typedesc[T]): T =
-  ## Returns a random integer in the range `low(T)..high(T)`.
-  result = cast[T](state.next)
+proc rand*[T: SomeInteger | enum](r: var Rand; t: typedesc[T]): T =
+  ## Returns a random element in the range `low(T)..high(T)`.
+  result = cast[T](r.next)
```

Comment:
This isn't correct for enums. `cast` will give a random value with the correct bit size, but not necessarily within the valid range for the enum.